### ISSUE TITLE
Date Field in Action popup closes the window

### ISSUE THREAD

**Issue by rorymcdaniel** [laravel/nova-issues#1834]
This looks to me the same as https://github.com/laravel/nova-issues/issues/1618

When I have a date field in an Action, choosing the date sometimes, but not always, closes the window. When this happens, the datepicker remains on the screen, but the Action window closes. Unscientific testing says the error happens 4 of 5 times.

Nova v2.0.9
Laravel 5.8.32
PHP 7.3.5
Debian 9.9 (via docker/Lando)
Mac Chrome 76.0.3809.100

![Screen Shot 2019-08-17 at 10 33 18 AM](https://user-images.githubusercontent.com/331850/63213452-7fbce480-c0da-11e9-8ac0-d0143304d912.png)
![Screen Shot 2019-08-17 at 10 21 58 AM](https://user-images.githubusercontent.com/331850/63212925-fce75a00-c0d8-11e9-88f6-4e6b29a94255.png)

After this happens, the following is in the console:
```Uncaught TypeError: Cannot read property 'value' of undefined
    at p.onChange (app.js?id=f8cc1c7f665f57468153:1)
    at ut (vendor.js?id=14d99f650c5d66f99957:1)
    at Object.W.l.close (vendor.js?id=14d99f650c5d66f99957:1)
    at J (vendor.js?id=14d99f650c5d66f99957:1)
    at HTMLDocument.<anonymous> (vendor.js?id=14d99f650c5d66f99957:1)```

**Comment by jbrooksuk** [comment]
I was finally able to recreate it just by selecting lots of dates.

**Comment by seanmangar** [comment]
Having the same issue, it's a bit erratic

**Comment by seanmangar** [comment]
Having a similar issue as well, though I get this in the console
```
app.js?id=f8cc1c7f665f57468153:1 Uncaught TypeError: Cannot read property 'value' of undefined
    at p.onChange (app.js?id=f8cc1c7f665f57468153:1)
    at ut (vendor.js?id=14d99f650c5d66f99957:formatted:21416)
    at Object.W.l.close (vendor.js?id=14d99f650c5d66f99957:formatted:21270)
    at J (vendor.js?id=14d99f650c5d66f99957:formatted:20957)
    at HTMLDocument.<anonymous> (vendor.js?id=14d99f650c5d66f99957:formatted:20779)
```

**Comment by jbrooksuk** [comment]
After a little bit of digging, it seems that the issue is caused within the `onChange` method. Occasionally, and for no obvious reason, `event` parameter is empty which results in `this.$refs.datePicker.value` not working either. I'd think that this would be a Flatpickr issue, but I can't see any similar issues where the solution works for us (after trying a handful).

Any thoughts?

**Comment by demency** [comment]
@jbrooksuk 
Apparently the modal is above the date picker, so if you click on any date this takes it as if the user had made a click outside the date picker... (So, value null is correct).

In my project I had to make a CSS modification.

**Comment by jbrooksuk** [comment]
> In my project I had to make a CSS modification.

Care to share the changes you made?

**Comment by demency** [comment]
@jbrooksuk 
It was not a modification to the CSS :|

The date picker component emitted an event when certain items were selected. I discovered it by debugging the events that were generated in DevTools. For some reason, the modal component allows a property to list the elements in a white list, thus avoiding the emission of the event by the component.

To solve the problem I added the classes that must be passed in the white list in the ConfirmActionModal.vue file.

Thus the component does not emit the modal closing signal.
```
<template>
    <modal
        data-testid="confirm-action-modal"
        tabindex="-1"
        role="dialog"
        @modal-close="handleClose"
        :classWhitelist="[
            'flatpickr-current-month',
            'flatpickr-next-month',
            'flatpickr-prev-month',
            'flatpickr-weekday',
            'flatpickr-weekdays',
            'flatpickr-calendar'
        ]"
    >
```

**Comment by jbrooksuk** [comment]
Thanks @demency! I've just tested this and am now unable to replicate the issue. I'll make a PR.

**Comment by RemiChalret** [comment]
Hi @jbrooksuk, any information on this issue?
The issue has been closed but we still have this problem on version 2.6.1 😕

**Comment by MartinRdw** [comment]
Hi @jbrooksuk, any information on this issue?
The issue has been closed but we still have this problem on version 2.6.1 😕

**Comment by jbrooksuk** [comment]
> Hi @jbrooksuk, any information on this issue?
> The issue has been closed but we still have this problem on version 2.6.1 😕

Heh, no because we didn’t know it was broken. 

Can you open a new ticket, please.

**Comment by czemu** [comment]
I had the same problem and I solved it by adding the main flatpickr container class name to the `classWhitelist` attribute of the modal component.

Full example:
```
<modal classWhitelist="flatpickr-calendar">
   // Your modal content here
</modal>
```

**Comment by santilorenzo** [comment]
Seems to be fixed in version 2.12.0
